### PR TITLE
STAC-13160 Test splitting

### PR DESCRIPTION
**Pull request by Jacco**
### Step 1: Link to Jira issue

https://stackstate.atlassian.net/browse/STAC-13160

### Step 2: Description of changes

Last BIG MR (I promise)

168 files, but it is about 148 json files.

Apart from the test moving here is an extensive list of other things that had to be changed in the process:
- lambda.py exception handling

Essentially it is splitting up the tests in to tests per API.
Also API response file are move to directories per API.
Test files have a different name strategy: methodname_hash.json
The hash is constructed from parameters and region. This way you can have different responses for same API call depending on parameters / region (for multi account support we'll introduce an options to use account_id as parent directory)
Also the testfile will have ResponseMetadata mentioning parameters and region used.
Lots of logic has moved into conftest.py

### Step 3: Did you add / update tests for your changes in the right area?
- [x] Unit/Component test
- [ ] Integration test	

### Step 4: I'm confident that everything is properly tested:
I got a PO / QA Approval by:
- [ ] Name

### Step 5: Can we ship this feature to production?
- [ ] Yes, I'm proud of my work. Ship it! :ship:

### CODE REVIEW

File: aws_topology/tests/test_cloudformation.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        )
+
+        self.assert_has_relation(
+            relations,
+            stack1,
+            stack2,
+            type="child of"
+        )
+
+        self.assertEqual(len(components), self.components_checked)
```
Comment (zandrewitte) [inline aws_topology/tests/test_cloudformation.py:66]: I wonder if we should specify an error message here (for all these len tests) that just prints out the actual components so you can use that as a starting point for debugging

File: aws_topology/tests/conftest.py [hunk 2]
```diff
@@ -64,3 +75,218 @@ def init_config_override(request):
...
+
+    def tearDown(self):
+        self.patcher.stop()
+
+    def assert_executed_ok(self):
+        service_checks = aggregator.service_checks(self.check.SERVICE_CHECK_EXECUTE_NAME)
+        self.assertGreater(len(service_checks), 0)
+        self.assertEqual(service_checks[0].status, AgentCheck.OK, service_checks[0].message)
+
+    def assert_has_component(self, components, id, type, checks={}):
```
[the remark this answers is not in the document]
Comment (Jacco) [reply aws_topology/tests/conftest.py:251]: Need a little pointer here. What is the stub for?

File: aws_topology/tests/conftest.py [hunk 3]
```diff
@@ -64,3 +75,218 @@ def init_config_override(request):
...
+        for relation in relations:
+            if relation["source_id"] == source_id and relation["target_id"] == target_id:
+                rel = relation
+                break
+        if rel is None:
+            print("Relations found:")
+            for relation in relations:
+                print("{} <-> {}".format(relation.get("source_id"), relation.get("target_id")))
+        self.assertIsNotNone(rel, "Relation expected source_id={} target_id={}".format(source_id, target_id))
+        if type:
```
[the remark this answers is not in the document]
Comment (Jacco) [reply aws_topology/tests/conftest.py:279]: I can do that :-) But the types were not checked in original

File: aws_topology/tests/test_cloudformation.py (hunk 1 above)
Comment (Jacco) [reply aws_topology/tests/test_cloudformation.py:66]: Was thinking the same thing. Other improvement I was thinking about: check the ids so we cam display the unchecked components/relations

**Review by zandrewitte — APPROVED** [review]

**Pull request merged**